### PR TITLE
Bugfix/ b1]  variables should be declared explicitly]

### PR DESCRIPTION
**Pull request by nyneshreddy**
**What's in the PR**
* The blocker issues have been solved
Declaring a variable inside a function or for loop without using the let, const or var keywords leads to the creation of global variables. The global namespace is kind of cluttered on browsers and creating a global variable leads to easy access to the variables and functions creating security concerns. This can be avoided by using let, const or var keywords. Security can be implemented by using "use strict"; method.
**How to test manually**
* ...

**Automatic testing**

] PR includes unit tests

### CODE REVIEW

File: .sonarcloud.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (munterkalmsteiner) [inline .sonarcloud.properties]: Sonar files should not be added to the repository.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+language: java
```
Comment (munterkalmsteiner) [inline .travis.yml]: Why is this necessary?

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full.js [hunk 3]
```diff
@@ -1250,7 +1250,7 @@
 //      } else {
 //        return this.adder.hide();
 //      }
-      annotation = this.setupAnnotation(this.createAnnotation());
+      var annotation = this.setupAnnotation(this.createAnnotation());
```
Comment (munterkalmsteiner) [inline inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full.js:1253]: Ok

File: inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/chart/resources/chart.js [hunk 4]
```diff
@@ -39,7 +39,7 @@ function updateLearningCurveDiagram(arrayOfLearningCurves) {
 
     // if we just have one value per data-row, we cannot visualize a step
     var plotTypes = {};
-	for (i = 0; i < arrayOfLearningCurves.length; i++) {
+	for (var i = 0; i < arrayOfLearningCurves.length; i++) {
```
Comment (munterkalmsteiner) [inline inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/chart/resources/chart.js:42]: Ok

File: inception-ui-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/search/sidebar/SearchAnnotationSidebar.java [hunk 5]
```diff
@@ -18,7 +18,7 @@
 package de.tudarmstadt.ukp.inception.app.ui.search.sidebar;
 
 import static de.tudarmstadt.ukp.clarin.webanno.api.annotation.util.WebAnnoCasUtil.getAddr;
-import static de.tudarmstadt.ukp.clarin.webanno.api.annotation.util.WebAnnoCasUtil.selectSingleFsAt;
+import static org.apache.uima.fit.util.CasUtil.selectSingleAt;
```
Comment (munterkalmsteiner) [inline inception-ui-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/search/sidebar/SearchAnnotationSidebar.java]: Not sure why this is necessary? This changes the behavior of the program and has nothing to do with the overall intention of the pull request (explicitly declaring variables).

**Review by munterkalmsteiner — CHANGES_REQUESTED** [review]
Hi,
the pull request description does not explain what has been changed why. Also, what are the files referenced under "automatic testing"?

There are some changes that seem not to be necessary or related to the intended improvement.

File: inception-ui-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/search/sidebar/SearchAnnotationSidebar.java (hunk 5 above)
Comment (nyneshreddy) [reply inception-ui-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/search/sidebar/SearchAnnotationSidebar.java]: This change has been removed.

File: .sonarcloud.properties (hunk 1 above)
Comment (nyneshreddy) [reply .sonarcloud.properties]: Removed the unnecessary file.

File: .travis.yml (hunk 2 above)
Comment (nyneshreddy) [reply .travis.yml]: Removed the unnecessary file.

**Review by munterkalmsteiner — APPROVED** [review]
Looks good!
